pandas/tests/strings/test_get_dummies.py

```diff
@@ -98,30 +91,19 @@ def test_get_dummies_with_pyarrow_dtype(any_string_dtype, dtype):
 
 
 # GH#47872
-@pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
 def test_get_dummies_with_str_dtype(any_string_dtype):
     s = Series(["a|b", "a|c", np.nan], dtype=any_string_dtype)
-    result = s.str.get_dummies("|", dtype=str)
-    expected = DataFrame(
-        [["T", "T", "F"], ["T", "F", "T"], ["F", "F", "F"]],
-        columns=list("abc"),
-        dtype=str,
-    )
-    tm.assert_frame_equal(result, expected)
+    with pytest.raises(
+        ValueError, match="string dtype not supported, please use a numeric dtype"
+    ):
+        s.str.get_dummies("|", dtype=str)
 
 
 # GH#47872
 @td.skip_if_no("pyarrow")
 def test_get_dummies_with_pa_str_dtype(any_string_dtype):
     s = Series(["a|b", "a|c", np.nan], dtype=any_string_dtype)
-    result = s.str.get_dummies("|", dtype="str[pyarrow]")
-    expected = DataFrame(
-        [
-            ["true", "true", "false"],
-            ["true", "false", "true"],
-            ["false", "false", "false"],
-        ],
-        columns=list("abc"),
-        dtype="str[pyarrow]",
-    )
-    tm.assert_frame_equal(result, expected)
+    with pytest.raises(
+        ValueError, match="string dtype not supported, please use a numeric dtype"
+    ):
+        s.str.get_dummies("|", dtype="str[pyarrow]")
```

Comment:
I think the test above is sufficient (so keep `test_get_dummies_with_str_dtype` but can remove `test_get_dummies_with_pa_str_dtype`